### PR TITLE
Feat/tutorial provider

### PR DESCRIPTION
**Pull request by aslakihle**
[AB#454904](https://dev.azure.com/Equinor/4319082b-29ba-4111-a20b-c04f55c7e7c6/_workitems/edit/454904)

### CODE REVIEW

File: src/providers/TutorialProvider/TutorialProvider.test.tsx [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+          router={getMemoryRouter({
+            tutorial,
+            withWrongCustomComponentKeyString: true,
+          })}
+        />
+      );
+      expect(spy).toHaveBeenCalledTimes(1);
+
+      const errorDialogText = screen.getByText(
+        /There was a problem starting this tutorial./i
```
[the remark this answers is not in the document]
Comment (aslakihle) [reply src/providers/TutorialProvider/TutorialProvider.test.tsx:435]: The reason is probably just that i copied the text c:

File: src/providers/TutorialProvider/TutorialProviderStory.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      imgUrl: 'https://placehold.co/800x200/png',
+    },
+    {
+      key: 'anotherCustomKey',
+    },
+  ],
+};
+
+const handleOnClick = () => {
+  console.log('Nevermind');
```
[the remark this answers is not in the document]
Comment (aslakihle) [reply src/providers/TutorialProvider/TutorialProviderStory.tsx:97]: It was kind of a joke in the story with the "U cant touch this" button 😂 
But i kept it to visualize that click events dont happen behind the highlighter during a tutorial. 
Could maybe make it even more clear (something visual on the page) to make the point without having to open console?

[the remark this answers is not in the document]
Comment (mariush2) [reply src/providers/TutorialProvider/TutorialProviderStory.tsx]: Ah ok, maybe you could use alert('') instead?

[the remark this answers is not in the document]
Comment (aslakihle) [reply src/providers/TutorialProvider/TutorialProviderStory.tsx]: Ended up just toggling between two variants on the button itself on click

**Review by mariush2 — APPROVED** [review]

**Review by mariush2 — APPROVED** [review]

**Pull request merged**